bugsnag-android-core/src/main/java/com/bugsnag/android/ThreadState.kt

```diff
@@ -5,37 +5,29 @@ import java.io.IOException
 /**
  * Capture and serialize the state of all threads at the time of an exception.
  */
-internal class ThreadState : JsonStream.Streamable {
-
-    private val threads: List<Thread>
-
-    constructor(threads: List<Thread>) {
-        this.threads = threads
-    }
-
-    constructor(
-        config: ImmutableConfig,
-        currentThread: java.lang.Thread,
-        stackTraces: MutableMap<java.lang.Thread, Array<StackTraceElement>>,
-        exc: Throwable?
-    ) {
+internal class ThreadState @JvmOverloads constructor
+    (
+    config: ImmutableConfig,
+    excStacktrace: Array<StackTraceElement>,
+    currentThread: java.lang.Thread = java.lang.Thread.currentThread(),
+    stackTraces: MutableMap<java.lang.Thread, Array<StackTraceElement>> = java.lang.Thread.getAllStackTraces()
+) : JsonStream.Streamable {
+
+    internal val threads: MutableList<Thread>
+
+    init {
+        // unhandled errors use the exception trace
         // API 24/25 don't record the currentThread, add it in manually
         // https://issuetracker.google.com/issues/64122757
-        if (!stackTraces.containsKey(currentThread)) {
-            stackTraces[currentThread] = currentThread.stackTrace
-        }
-        if (exc != null) { // unhandled errors use the exception trace
-            stackTraces[currentThread] = exc.stackTrace
-        }
+        stackTraces[currentThread] = excStacktrace
```

Comment:
Can you guarantee that for non-API-24/25 stracktraces, it has no impact to reassign the existing one? I would rather see the original guard still in place to check the key.

Reply:
I have reverted this change to use the original guard.